model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -401,6 +425,86 @@ async def create_deepspeed_bundle(
                 )
             ).model_bundle_id
 
+    async def create_vllm_bundle(
+        self,
+        user: User,
+        model_name: str,
+        framework_image_tag: str,
+        endpoint_unique_name: str,
+        num_shards: int,
+        checkpoint_path: Optional[str],
+    ):
+        command = []
+
+        max_num_batched_tokens = 2560  # vLLM's default
+        if "llama-2" in model_name:
+            max_num_batched_tokens = 4096  # Need to be bigger than model's context window
+
+        assert checkpoint_path is not None, "Checkpoint path must be provided for vLLM"
+
+        if checkpoint_path.startswith("s3://"):
+            base_path = checkpoint_path.split("/")[-1]
+            final_weights_folder = "model_files"
+            subcommands = []
+
+            s5cmd = "./s5cmd"
+
+            if base_path.endswith(".tar"):
+                # If the checkpoint file is a tar file, extract it into final_weights_folder
+                subcommands.extend(
+                    [
+                        f"{s5cmd} cp {checkpoint_path} .",
```

Comment:
Probably fine for now, but if we have 3 frameworks, we might want to refactor this s5cmd piece into a common component.

Reply:
sure